mautic/mautic
# Do not submit emails validation

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | x
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
A coporation marketer requires do not submit form for certain emails. This PR added new Do Not Submit emails list to configuration and optional validation  for this list
This feature is offered by [Salesforce Pardot](https://www.pardot.com/blog/preventing-competitors-from-accessing-content-on-form-submission/).

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Go to Configuration > Form Settings > Do not submit these emails
- add email example@domain.tld (should blocked this email)
- add domain with * (*@gmail.com) - should blocked all gmail emails
3. Create form with email enable Do Not Submit  validation. Also try customization of validation message
4. Then try submit with emails
5. Form should return error message

Screenshot

![image](https://user-images.githubusercontent.com/462477/61881340-c4e46100-aef6-11e9-9693-07634e7af9da.png)

![image](https://user-images.githubusercontent.com/462477/61882400-c151d980-aef8-11e9-85b6-b000fac75e86.png)
